node/src/components/chain_synchronizer/operations.rs

```diff
@@ -1450,19 +1450,19 @@ async fn fetch_finality_signatures_by_block_header(
                                 validator_weights,
                             )?;
                             sig_collector.add(signatures);
-                        }
 
-                        if sig_collector.check_if_sufficient(
-                            validator_weights,
-                            ctx.config.finality_threshold_fraction(),
-                        ) {
-                            info!(
-                                ?block_header_hash,
-                                height = block_header.height(),
-                                ?era_for_validators_retrieval,
-                                "fetched sufficient finalty signatures"
-                            );
-                            break;
+                            if sig_collector.check_if_sufficient(
```

Comment:
If `signatures` are already known, we would still do the check unnecessarily.

Reply:
What do you mean by "known" here?

Do you think we should check if `add` actually added any "new" signatures and only then call `check_if_sufficient`?

Reply:
Yes.